chemprop/utils.py

```diff
@@ -528,3 +528,81 @@ def save_smiles_splits(data_path: str,
 
     with open(os.path.join(save_dir, 'split_indices.pckl'), 'wb') as f:
         pickle.dump(all_split_indices, f)
+
+
+def update_prediction_args(predict_args: PredictArgs,
+                           train_args: TrainArgs,
+                           missing_to_defaults: bool = True,
+                           for_molecule_fingerprint: bool = False) -> None:
+    """
+    Updates prediction arguments with training arguments loaded from a checkpoint file.
+    If an argument is present in both, the predicion argument will be used.
+
+    Also raises errors for situations where the prediction arguments and training arguments
+    are different but must match for proper function.
+
+    :param predict_args: The :class:`~chemprop.args.PredictArgs` object containing the arguments to use for making predictions.
+    :param train_args: The :class:`~chemprop.args.TrainArgs` object containing the arguments used to train the model previously.
+    :param missing_to_defaults: Whether to replace missing training arguments with the current defaults for :class: `~chemprop.args.TrainArgs`.
+        This is used for backwards compatibility.
+    :param for_molecule_fingerprint: Indicates whether this argument update is being used for molecule fingerprint generation,
+        in which case molecule features are not needed, even if used in the model training.
+    """
+    for key, value in vars(train_args).items():
+        if not hasattr(predict_args, key):
+            setattr(predict_args, key, value)
+
+    if missing_to_defaults:
+        override_defaults = dict((
+            ('bond_features_scaling',False),
+            ('no_bond_features_scaling',True),
+            ('atom_descriptors_scaling',False),
+            ('no_atom_descriptors_scaling',True),
+        ))
```

Comment:
I think the override dictionary is a good solution. However, I think there is a line missing where you actually apply this, or am I missing something? There should probably a if/else clause somewhere in line 563-565?

Reply:
Ah! I got ahead of myself here. Whoops, got too focused on finding the arguments and left out the part where it actually overrides.

This is now fixed and tested!

Reply:
Nice!